src/graph.cpp

```diff
@@ -145,15 +170,36 @@ void Graph::kruskal(){
  * Time Complexity: O(|V|+|E|)
  * @param source - Vertex where the DFS starts
  */
-void Graph::dfsKruskalPath(Vertex *source){
-    v->setVisited(true);
-    for (Edge *e : v->getAdj()){
-        if(!e->getDest()->isVisited() && e->isSelected()){
-            e->getDest()->setPath(e);
-            dfsKruskalPath(e->getDest());
+void Graph::preorderMSTTraversal(Vertex *source){
+    source->setVisited(true);
+    if (tour.second.size() != 0){
+        tour.first += findEdge((*(tour.second).rend())->getId(), source->getId())->getDist();
+    }
+    tour.second.push_back(source);
+    for (Edge *e : source->getAdj()){
+        bool unvisited = !(e->getDest()->isVisited());
+        bool selected = e->isSelected();
+        //bool connected_to_last = isConnectable(e->getDest());
+        if(unvisited && selected /*&& connected_to_last*/){
+            preorderMSTTraversal(e->getDest());
         }
     }
-    return;
+}
+
+/**
+ * Calculates an approximation of the TSP, using the triangular approximation heuristic
+ * Time Complexity: //TODO
+ */
+void Graph::triangularTSPTour(){
+    /*
+    -Build MST
+    -Get pre-order of the mst as vector //!DFS?
+    -Iterate that order getting total dist
+    */
+    kruskal();
+    tour = {0,{}};
+    preorderMSTTraversal(*(vertexSet.begin()));
+    return; //results are updated in tour
```

Comment:
We are missing here adding the edge between the last vertex back to the first one